From 1838bb17b1beda6857a4d6498c65daa4f52c8dd5 Mon Sep 17 00:00:00 2001
From: Alex Leibovich <alexl@marvell.com>
Date: Wed, 26 May 2021 09:40:29 -0400
Subject: [PATCH 1560/1921] net: mvpp2: Add NETIF_F_HW_VLAN_CTAG_FILTER to
 dev->features

Adding NETIF_F_HW_VLAN_CTAG_FILTER to dev->features
in order to solve VLAN0 crash issue.

Signed-off-by: Alex Leibovich <alexl@marvell.com>
Change-Id: I2580005d9462ee89c6d86bc8c4b3ee649a790519
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/52721
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
(cherry picked from commit 9a4eab2c295431274d749987244ac830b1b7d0bc)
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/52730
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c b/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
index 1bc813618b6c..709ce85b7c24 100644
--- a/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
+++ b/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
@@ -6826,10 +6826,9 @@ static int mvpp2_port_probe(struct platform_device *pdev,
 	}
 
 	features = NETIF_F_SG | NETIF_F_IP_CSUM | NETIF_F_IPV6_CSUM |
-		   NETIF_F_TSO;
+		   NETIF_F_TSO | NETIF_F_HW_VLAN_CTAG_FILTER;
 	dev->features = features | NETIF_F_RXCSUM;
-	dev->hw_features |= features | NETIF_F_RXCSUM | NETIF_F_GRO |
-			    NETIF_F_HW_VLAN_CTAG_FILTER;
+	dev->hw_features |= features | NETIF_F_RXCSUM | NETIF_F_GRO;
 
 	if (mvpp22_rss_is_supported(port))
 		dev->hw_features |= NETIF_F_RXHASH;
-- 
2.31.1

